File: 2025_03/[프로그래머스 Lv.1] 음양 더하기/음양 더하기.js
function solution(absolutes, signs) {
  let result = 0
  for(let i = 0 ; i < absolutes.length ;i++){
    result += signs[i] ? absolutes[i] : -absolutes[i]
  }
  return result
}

// reduce 이용 풀이
function solution(absolutes, signs) {
  return absolutes.reduce((acc, val, i) => acc + (val * (signs[i] ? 1 : -1)), 0);
}

File: 2025_03/프로그래머스 - Lv.1/음양 더하기/음양 더하기.py
def solution(absolutes, signs):
    answer = 0
    for i in range(len(absolutes)):
        answer += absolutes[i] if signs[i] else -absolutes[i]
    return answer

# zip 이용 풀이
def solution(absolutes, signs):
    return sum(absolutes if sign else -absolutes for absolutes, sign in zip(absolutes, signs))
